testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -1000,6 +1000,14 @@ void testCastToInterval(CastType castType, SqlOperatorFixture f) {
         "cast(-5723 as interval minute(4))",
         "-5723",
         "INTERVAL MINUTE(4) NOT NULL");
+    f.checkScalar(
+        "cast(cast(1 as integer) as interval minute)",
+        "+1",
+        "INTERVAL MINUTE NOT NULL");
+    f.checkScalar(
```

Comment:
how about other interval types, such as `INTERVAL HOUR`?

Reply:
I've added test for HOUR, MONTH and SECOND intervals.